ql/experimental/math/particleswarmoptimization.cpp

```diff
@@ -17,10 +17,6 @@ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 FOR A PARTICULAR PURPOSE.  See the license for more details.
 */
 
-#include <ql/qldefines.hpp>
-
-#if BOOST_VERSION >= 104700
```

Comment:
As for qldefines, it is already included in the hpp file, so it was just oversight to do it again there.  
I think I was just too overzealous in trying to clean up files. Now I see that I must have removed the guard in the cpp file because I thought that there was no include in it that needed the guard, but I see I must also have mistakenly removed it from the header file, meaning that it will indeed stop compilation with boost lower than 1.47. So I think the boost version guard needs to be added again to both the cpp and the hpp file. Sorry about that.